test/inductor/test_torchinductor.py

```diff
@@ -7969,6 +7970,48 @@ def fn(x):
 
         self.common(fn, (torch.randn((16, 32)),), check_lowp=False)
 
+    @requires_cuda()
+    @torch._inductor.config.patch("layout_optimization", True)
+    @torch._inductor.config.patch("keep_output_stride", False)
+    @config.patch(implicit_fallbacks=True)
+    def test_custom_op_fixed_layout(self):
+        import torch.library
+
+        mod = nn.Conv2d(3, 128, 1, stride=1, bias=False).cuda()
+        inp = torch.rand(2, 3, 128, 128, device="cuda")
+        expected_stride = mod(inp).stride()
```

Comment:
I expect `expected_stride` to be contiguous. Is that the case?

If so, we want to test with some non-contiguous strides as well. Things like `x = torch.randn(10), x[::2]`, `x = torch.randn(2, 3), x.t()`.

Reply:
Added a second test that takes channels_last and confirms the layout is channels last after. For non contiguous strides like [::2], we don't currently preserve strides(only the memory layout) but we can add it as a followup.